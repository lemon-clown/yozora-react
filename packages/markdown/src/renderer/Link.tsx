import type { Link } from '@yozora/ast'
import LinkRenderer from '@yozora/react-link'
import React, { useContext } from 'react'
import { YozoraMarkdownContext } from '../context/context'

export function YozoraLink(link: Link): React.ReactElement {
  const { renderYozoraNodes } = useContext(YozoraMarkdownContext)
  const { url, title, children } = link

  return (
    <LinkRenderer url={url} title={title}>
      {renderYozoraNodes(children)}
    </LinkRenderer>
  )
}

YozoraLink.displayName = 'YozoraLink'
export default YozoraLink
