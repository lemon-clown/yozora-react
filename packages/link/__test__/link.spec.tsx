import { render } from 'enzyme'
import React from 'react'
import Link from '../src'

const url = 'https://github.com/guanghechen'
const children = (
  <React.Fragment>
    Some link text
    <span>Some link text2</span>
  </React.Fragment>
)

describe('prop types', function () {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })
  })

  it('url is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<Link url={value}>{children}</Link>)
      }).toThrow(/The prop `url` is marked as required/i)
    }
  })

  describe('className is optional', function () {
    it('default', function () {
      const node = render(<Link url={url}>{children}</Link>)
      expect(node.hasClass('yozora-link')).toBeTruthy()
    })

    it('custom', function () {
      const node = render(
        <Link url={url} className="my-code">
          {children}
        </Link>,
      )
      expect(node.hasClass('yozora-link')).toBeTruthy()
      expect(node.hasClass('my-code')).toBeTruthy()
    })
  })

  it('style is optional', function () {
    const node = render(
      <Link url={url} style={{ color: 'orange' }}>
        {children}
      </Link>,
    )
    expect(node.css('color')).toEqual('orange')
  })
})

describe('snapshot', function () {
  it('basic', () => {
    const wrapper = render(
      <Link
        url={url}
        title="home"
        style={{ color: 'orange', fontSize: '16px' }}
      >
        {children}
      </Link>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
