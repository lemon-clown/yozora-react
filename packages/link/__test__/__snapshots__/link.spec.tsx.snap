// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot basic 1`] = `
<a
  class="yozora-link"
  href="https://github.com/guanghechen"
  rel="noopener,noreferrer"
  style="color:orange;font-size:16px"
  target="_blank"
  title="home"
>
  Some link text
  <span>
    Some link text2
  </span>
</a>
`;
